packages/data/addon/services/keg.ts

```diff
@@ -85,6 +91,51 @@ export default class KegService extends Service {
     return this.pushMany(type, [rawRecord], options).firstObject;
   }
 
+  /**
+   * Inserts a single record into the store
+   *
+   * @param type - name of the model type
+   * @param record - record to be inserted into the keg
+   * @param options - config object
+   */
+  insert(type: string, record: KegRecord, options: Partial<InsertOptions> = {}): KegRecord {
+    return this.insertMany(type, [record], options).firstObject as KegRecord;
+  }
+
+  /**
+   * Inserts an array of records into the store
+   *
+   * @param type - type name of the model type
+   * @param record - record to be inserted into the keg
+   * @param options - config object
+   */
+  insertMany(type: string, records: Array<KegRecord>, options: Partial<InsertOptions> = {}): EmberArray<KegRecord> {
+    const recordKeg = this._getRecordKegForType(type);
+    const idIndex = this._getIdIndexForType(type);
+    const namespace = options.namespace || this.defaultNamespace;
+    const identifierField = options?.identifierField || KegService.identifierField;
```

Comment:
```suggestion
    const identifierField = options.identifierField || KegService.identifierField;
```

Reply:
`options` is defaulted to `{}`

Reply:
right that's why the optional chaining isn't needed, right? Or if it is needed, it should be added to the line above where we access `options.namespace`

Reply:
ah, read your comment the other way.